modules/microshift-cni.adoc

```diff
@@ -0,0 +1,301 @@
+
+== Configuring OVN-Kubernetes
+An OVN-Kubernetes config file must be written to /etc/microshift/ovn.yaml. {product-title} will use default OVN-Kubernetes configuration values if an OVN-Kubernetes config file is not customized.
+//Q: what are the default values?
+
+.Supported OVN-Kubernetes configurations for {product-title}.
+[cols="6",options="header"]
+|===
+|Field
+|Required
+|Type
+|Default
+|Description
+|Example
+
+|`disableOVSInit`
+|N
+|bool
+|false
+|Skip configuring OVS bridge `br-ex` in `microshift-ovs-init.service`
+|true <1>
+
+|mtu
+|N
+|uint32
+|1400
+|MTU value used for the pods
+|1300
+|===
+
+<1> The OVS bridge is required. When `disableOVSInit` is true, OVS bridge `br-ex` must be configured manually.
+
+.Example `ovn.yaml` config file:
+//Q: this doesn't match the table above; should it have all six fields?
+[source, yaml]
+----
+disableOVSInit: true
+mtu: 1300
+Configuring Host
+OVS bridge
+----
+[IMPORTANT]
+When `disableOVSInit` is set to true in the `ovn.yaml` config file, the OVS bridge br-ex must be manually configured.
+
+=== Manually configuring OVS bridge br-ex
+.Procedure
+Manually configur the OVS bridge br-ex by running the following commands.
+//Q: Need SME review of each command's function.
+* Initiate OVS:
++
+[source, terminal]
+----
+$ sudo systemctl enable openvswitch --now
+----
+
+* Add the network bridge:
++
+[source, terminal]
+----
+$ sudo ovs-vsctl add-br br-ex
+----
+
+* Add the interface to the network bridge:
++
+[source, terminal]
+----
+$ sudo ovs-vsctl add-port br-ex <physical-interface-name>
+----
+The `<physical-interface-name>` is the network interface name where the node IP address is assigned.
+
+* Get the bridge up and running:
++
+[source, terminal]
+----
+$ sudo ip link set br-ex up
+----
+//Q: Is this last sentence actually for the step preceeding.
+* After `br-ex up` is running, assign the node IP address to `br-ex` bridge:
+
+[source, terminal]
+----
+$ sudo ...
+----
+//missing procedure/command here
+[NOTE]
+Adding a physical interface to `br-ex` bridge will disconnect the ssh connection on the node IP address.
+
+== Firewalld
+Explicitly allow the following OVN-Kubernetes traffic when the firewalld service is running:
+//Q: procedure?
+CNI pod to CNI pod::
+CNI pod to Host-Network pod
+Host-Network pod to Host-Network pod
+
+CNI pod::
+The Kubernetes pod that uses the CNI network
+
+Host-Network pod::
+The Kubernetes pod that uses host network
+
+.Procedure
+Insert the following firewall rules to allow the OVN-Kubernetes traffic:
+//Q: are these commands doing what we say? please review
+* Insert the IP range:
+[source, terminal]
+----
+$ sudo firewall-cmd --permanent --zone=trusted --add-source=10.42.0.0/16
+----
+
+* Insert the DNS server:
+[source, terminal]
+----
+$ sudo firewall-cmd --permanent --zone=trusted --add-source=169.254.169.1
+----
+
+* Reload the firewall rules:
++
+[source, terminal]
+----
+$ sudo firewall-cmd --reload
+----
+
```

Comment:
firewall-cmd should be changed into firewall-offline-cmd if configured from kickstart
(Shauna: kickstart is a scripting language used to put into "CDs"/RHEL images ... so when you plug them on the device/computer it will install automatically). i.e.: 

https://github.com/openshift/microshift/blob/main/docs/config/microshift-starter.ks#L67